AppCenter/AppCenter/Internals/AppDelegate/MSAppDelegateForwarder.m

```diff
@@ -220,12 +228,21 @@ + (IMP)swizzleOriginalSelector:(SEL)originalSelector
       warningMsg =
           [NSString stringWithFormat:@"This selector is not supported when already implemented. %@", remediationMsg];
     }
+
+    // Check if this selector is deprecated.
+    if ([self.deprecatedSelectors.allValues containsObject:originalSelectorStr]) {
+
+      // This deprecated selector is also implemented in the original delegate.
+      [self.deprecatedSelectorsImplementationCheck setObject:@(YES) forKey:originalSelectorStr];
+    }
   } else if (![originalClass instancesRespondToSelector:originalSelector]) {
 
-    // Check for deprecation.
+    // Check if any associated deprecated selector is implemented by the original delegate.
     NSString *deprecatedSelectorStr = self.deprecatedSelectors[originalSelectorStr];
     if (deprecatedSelectorStr &&
-        [originalClass instancesRespondToSelector:NSSelectorFromString(deprecatedSelectorStr)]) {
+        ([self.deprecatedSelectorsImplementationCheck[deprecatedSelectorStr] boolValue] ||
```

Comment:
Doesn't look `deprecatedSelectorsImplementationCheck` has values for keys.

Reply:
?